src/test/java/healper/impl/SmartContractImpl.java

```diff
@@ -0,0 +1,146 @@
+                palauProperties.getTrigger().getRegisterBondId(),
+                palauProperties.getTrigger().getRegisterBondWasm());
+    }
+
+    @Override
+    public SignedTransaction deployRegisterBuyBondsTrigger(String activationCode) {
+        log.info("""
+                Deploying smart contracts:
+                {}
+                """,
+                palauProperties.getTrigger().getBuyBondWasm());
+
+        return registerBuyBondsTrigger(
+                palauProperties.getTrigger().getBuyBondId(),
+                palauProperties.getTrigger().getBuyBondWasm());
+    }
+
+    @Override
+    public SignedTransaction deployRegisterRedeemBondsTrigger(String activationCode) {
+        log.info("""
+                Deploying smart contracts:
+                {}
+                """,
+                palauProperties.getTrigger().getRedeemBondWasm());
+
+        return registerRedeemBondsTrigger(
+                palauProperties.getTrigger().getRedeemBondId(),
+                palauProperties.getTrigger().getRedeemBondWasm());
+    }
+
+    @SneakyThrows
+    private SignedTransaction registerBuyBondsTrigger(String buyBondId, String buyBondWasm) {
+        final var triggerId = new TriggerId(null, ExtensionsKt.asName(buyBondId));
+
+        final var wasm = Objects
+                .requireNonNull(getClass().getClassLoader().getResource(buyBondWasm))
+                .openConnection()
+                .getInputStream()
+                .readAllBytes();
+
+        final var filter = Filters.INSTANCE.data(
+                EntityFilters.INSTANCE.byAccount(
+                        null,
+                        new AccountEventFilter.ByMetadataInserted()));
+
+        return buildAndSubmitRegisterTriggerTransaction(triggerId, wasm, filter);
+    }
+
+    @SneakyThrows
+    private SignedTransaction registerRedeemBondsTrigger(String redeemBondId, String redeemBondWasm) {
+        final var triggerId = new TriggerId(null, ExtensionsKt.asName(redeemBondId));
+
+        final var wasm = Objects
+                .requireNonNull(getClass().getClassLoader().getResource(redeemBondWasm))
+                .openConnection()
+                .getInputStream()
+                .readAllBytes();
+
+        final var filter = Filters.INSTANCE.data(
+                EntityFilters.INSTANCE.byAccount(
+                        null,
+                        new AccountEventFilter.ByMetadataInserted()));
+
+        return buildAndSubmitRegisterTriggerTransaction(triggerId, wasm, filter);
+    }
+
+    @SneakyThrows
+    private SignedTransaction registerRegisterBondTrigger(String registerBondId, String registerBondWasm) {
+        final var triggerId = new TriggerId(null, ExtensionsKt.asName(registerBondId));
+
+        final var filter = buildFilterByTriggerMetadataInserted(triggerId);
+
+        final var wasm = Objects
+                .requireNonNull(getClass().getClassLoader().getResource(registerBondWasm))
+                .openConnection()
```

Comment:
Shouldn't the connection be closed after?

Reply:
it was closed automatically